sleap_io/io/nwb.py

```diff
@@ -0,0 +1,282 @@
+        .swaplevel(0, -1, axis=1)  # video_path on top while x, y score on bottom
+        .sort_index(axis=1)  # Better format for columns
+        .sort_index(axis=0)  # Sorts by frames
+    )
+
+    return labels_tidy_df
+
+
+def write_labels_to_nwb(labels: Labels, nwbfile_path: str):
+    """Write labels to an nwb file and save it to the nwbfile_path given
+
+    Args:
+        labels (Labels): A general label object
+        nwbfile_path (str): The path where the nwb file is to be written
+    """
+
+    session_description: str = "Processed SLEAP pose data"
+    session_start_time = datetime.datetime.now(datetime.timezone.utc)
+    identifier = "identifier"
+
+    nwbfile = NWBFile(
+        session_description=session_description,
+        identifier=identifier,
+        session_start_time=session_start_time,
+    )
+
+    nwbfile = append_labels_data_to_nwb(labels, nwbfile)
+
+    with NWBHDF5IO(str(nwbfile_path), "w") as io:
+        io.write(nwbfile)
+
+
+def append_labels_data_to_nwb(labels: Labels, nwbfile: NWBFile) -> NWBFile:
+    """
+
+    Args:
+        labels (Labels): A general labels object
+        nwbfile (NWBFile): And in-memory nwbfile where the data is to be appended.
+
+    Returns:
+        NWBFile: An in-memory nwbfile with the data from the labels object appended.
+    """
+
+    labels_data_df = _extract_predicted_instances_data(labels)
+
+    # For every video create a processing module
+    for video_index, video in enumerate(labels.videos):
+
+        video_path = Path(video.filename)
+        processing_module_name = f"SLEAP_VIDEO_{video_index:03}_{video_path.stem}"
+        nwb_processing_module = get_processing_module_for_video(
+            processing_module_name, nwbfile
+        )
+
+        name_of_tracks_in_video = (
+            labels_data_df[video.filename]
+            .columns.get_level_values("track_name")
+            .unique()
+        )
+
+        # For every track in that video create a PoseEstimation container
+        for track_index, track_name in enumerate(name_of_tracks_in_video):
+            pose_estimation_container = build_pose_estimation_container_for_track(
+                labels_data_df, labels, track_name, video
+            )
+            nwb_processing_module.add(pose_estimation_container)
+
+    return nwbfile
+
+
+def get_processing_module_for_video(
+    processing_module_name: str, nwbfile: NWBFile
+) -> ProcessingModule:
+    """Auxiliary function to create a processing module.
+
+    Checks for the processing module existence and creates if not available.
+
+    Args:
+        processing_module_name (str): The name of the processing module.
+        nwbfile (NWBFile): The nwbfile to attach the processing module to.
+
+    Returns:
+        ProcessingModule: An nwb processing module with the desired name.
+    """
+    if processing_module_name in nwbfile.processing:
+        nwb_processing_module = nwbfile.processing[processing_module_name]
```

Comment:
So this is for appending to an existing file? We could get coverage here by saving out one file and then saving out to it again under a different `processing_module_name`

Reply:
I tried a [ternary operator ](https://docs.python.org/3.10/reference/expressions.html?highlight=ternary#conditional-expressions)approach here in the last commit. We can still add the test if you want but I feel is an overkill.